tools/clang/test/SemaHLSL/v202x/conforming-literals/valid-literals.hlsl

```diff
@@ -0,0 +1,247 @@
+// RUN: %dxc -T lib_6_3 -HV 202x -verify %s
+// RUN: %dxc -T lib_6_3 -HV 202x -enable-16bit-types -verify %s
+// RUN: %dxc -T lib_6_3 -HV 2021 -verify %s
+// RUN: %dxc -T lib_6_3 -HV 2021 -enable-16bit-types -verify %s
+
+#if __HLSL_VERSION <= 2021
+// expected-no-diagnostics
+#endif
+
+template <typename T, typename U>
+struct is_same {
+  static const bool value = false;
+};
+
+template <typename T>
+struct is_same<T, T> {
+  static const bool value = true;
+};
+
+bool B; // Used for ternary operator tests below
+
+#if !defined(__HLSL_ENABLE_16_BIT)
+min16int i16;
```

Comment:
Would these tests all still pass if `-enable-16bit-types` was getting ignored?

Reply:
I mean... kinda by design. I wrote the test so it covers both when the flag is passed and when it isn't.

If the flag is completely ignored, this test would still pass, but a lot of other tests which test behaviors of that flag specifically would fail.